tests/TestEntry.cpp

```diff
@@ -173,20 +175,90 @@ void TestEntry::testResolveUrlPlaceholders()
     QString userinfo("user:pw"); // User information of the entry URL.
     QString username("user"); // User name of the entry URL.
     QString password("pw"); // Password of the entry URL.
-    QString fragment("fragment"); // Password of the entry URL.
-
-    QCOMPARE(entry->resolvePlaceholder("{URL:RMVSCM}"), rmvscm);
-    QCOMPARE(entry->resolvePlaceholder("{URL:WITHOUTSCHEME}"), rmvscm);
-    QCOMPARE(entry->resolvePlaceholder("{URL:SCM}"), scm);
-    QCOMPARE(entry->resolvePlaceholder("{URL:SCHEME}"), scm);
-    QCOMPARE(entry->resolvePlaceholder("{URL:HOST}"), host);
-    QCOMPARE(entry->resolvePlaceholder("{URL:PORT}"), port);
-    QCOMPARE(entry->resolvePlaceholder("{URL:PATH}"), path);
-    QCOMPARE(entry->resolvePlaceholder("{URL:QUERY}"), query);
-    QCOMPARE(entry->resolvePlaceholder("{URL:USERINFO}"), userinfo);
-    QCOMPARE(entry->resolvePlaceholder("{URL:USERNAME}"), username);
-    QCOMPARE(entry->resolvePlaceholder("{URL:PASSWORD}"), password);
-    QCOMPARE(entry->resolvePlaceholder("{URL:FRAGMENT}"), fragment);
+    QString fragment("fragment"); // Fragment of the entry URL.
 
-    delete entry;
+    QCOMPARE(entry.resolvePlaceholder("{URL:RMVSCM}"), rmvscm);
+    QCOMPARE(entry.resolvePlaceholder("{URL:WITHOUTSCHEME}"), rmvscm);
+    QCOMPARE(entry.resolvePlaceholder("{URL:SCM}"), scm);
+    QCOMPARE(entry.resolvePlaceholder("{URL:SCHEME}"), scm);
+    QCOMPARE(entry.resolvePlaceholder("{URL:HOST}"), host);
+    QCOMPARE(entry.resolvePlaceholder("{URL:PORT}"), port);
+    QCOMPARE(entry.resolvePlaceholder("{URL:PATH}"), path);
+    QCOMPARE(entry.resolvePlaceholder("{URL:QUERY}"), query);
+    QCOMPARE(entry.resolvePlaceholder("{URL:USERINFO}"), userinfo);
+    QCOMPARE(entry.resolvePlaceholder("{URL:USERNAME}"), username);
+    QCOMPARE(entry.resolvePlaceholder("{URL:PASSWORD}"), password);
+    QCOMPARE(entry.resolvePlaceholder("{URL:FRAGMENT}"), fragment);
+}
+
+void TestEntry::testResolveRecursivePlaceholders()
+{
+    Database db;
+    Group* root = db.rootGroup();
+
+    Entry* entry1 = new Entry();
```

Comment:
You used a stack variable for the test above, you can do the same here.

Reply:
@phoerious can I not use stack variables? The database will still destroy all records.

Reply:
You mean free store variables? Yes, you are right. I actually didn't mark it last time exactly for that reason. Forgot about it this time. All fine. Just rebase to current develop and I can merge this.